Packs/PrismaCloudCompute/Integrations/PaloAltoNetworks_PrismaCloudCompute/PaloAltoNetworks_PrismaCloudCompute.py

```diff
@@ -253,9 +268,522 @@ def fetch_incidents(client):
+        api_limit_call = 50
+
+        command_args = kwargs.get('args')
+        offset = arg_to_number(arg=command_args.get('offset'), arg_name='offset')  # type:ignore
+        if offset < 0:  # type:ignore
+            offset = 0
+        command_args['offset'] = offset  # type:ignore
+
+        limit = arg_to_number(arg=command_args.get('limit'), arg_name='limit')  # type:ignore
+        if limit > api_limit_call:  # type:ignore
+            limit = api_limit_call  # type:ignore
+        command_args['limit'] = limit  # type:ignore
+
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+def parse_date_string_format(
+    date_string: str, date_string_format: str = '%Y-%m-%dT%H:%M:%S.%fZ', new_format: str = "%B %d, %Y %H:%M:%S %p"
+) -> str:
+    """
+    Parses a date string format to a different date string format.
+
+    Args:
+        date_string (str): the date in string representation.
+        date_string_format (str): the current format of the string date.
+        new_format (str): the new requested format for the date string.
+
+    Returns:
+        str: date as a new format, in case of a failure returns the original date string.
+    """
+    try:
+        return parse_date_string(date_string=date_string, date_format=date_string_format).strftime(new_format)
+    except ValueError:
+        return date_string
+
+
+def perform_api_request(
+    client: PrismaCloudComputeClient, url_suffix: str, args: dict = None, method: str = 'GET', json_data: dict = None
+):
+    """
+    Perform api request for the PrismaCloudCompute client.
+
+    Args:
+        client (PrismaCloudComputeClient): prisma-cloud-compute client.
+        method (str): the api method type. e.g.: 'GET,POST,PUT'
+        url_suffix (str): url suffix of the base api url.
+        args (dict): any command arguments if exist.
+        json_data (dict): body request for the http request.
+
+    Returns:
+        list/dict: api response.
+    """
+    return client.api_request(
+        method=method, url_suffix=url_suffix, params=assign_params(**args) if args else {}, json_data=json_data
+    )
+
+
+def get_api_filtered_response(
+    client: PrismaCloudComputeClient, url_suffix: str, offset: int, limit: int, args: dict = None
+):
+    """
+    Filter the api response according to the offset/limit, used in case the api doesn't support limit/offset
+
+    Args:
+        client (PrismaCloudComputeClient): prisma-cloud-compute client.
+        url_suffix (str): url suffix of the base api url.
+        offset (int): the offset from which to begin listing the response.
+        limit (int): the maximum limit of records in the response to fetch.
+        args (dict): any command arguments if exist.
+
+    Returns:
+        list: api filtered response, empty list in case there aren't any records in the api response.
+    """
+    response = perform_api_request(client=client, url_suffix=url_suffix, args=args)
+
+    start = min(offset, len(response))
```

Comment:
need to improve handling:
![image](https://user-images.githubusercontent.com/31018228/144747062-cadb0722-4a4a-41f9-bf25-46acb341cc2a.png)

Reply:
well i went through most of the error scenarios that i could think off, ill take a deeper look and make sure we have valid responses